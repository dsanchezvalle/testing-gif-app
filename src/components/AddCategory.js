import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
        //console.log('Handle change activated')
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        //console.log('Handle Submit called', inputValue)
        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
