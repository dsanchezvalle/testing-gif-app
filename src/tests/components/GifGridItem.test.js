import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem'

describe('GifGridItem tests', ()=>{
    test('it should render GifGridItem properly', ()=>{
        const wrapper = shallow(<GifGridItem />);
        expect(wrapper).toMatchSnapshot();
    });
})