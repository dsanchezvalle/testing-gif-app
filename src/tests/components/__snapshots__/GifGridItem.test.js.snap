// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests it should render GifGridItem properly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p>
     
     
  </p>
</div>
`;
